pkg/devfile/adapters/common/command.go

```diff
@@ -39,8 +40,15 @@ func getCommandFromDevfile(data data.DevfileData, groupType common.DevfileComman
 	}
 
 	for _, command := range commands {
-		// exec.Group is a pointer, to avoid null pointer
-		if command.Exec.Group != nil && command.Exec.Group.Kind == groupType {
+		if command.Composite != nil && command.Composite.Group != nil && command.Composite.Group.Kind == groupType {
```

Comment:
I believe we are overpopulating this function with complex logic,  @johnmcollier is it possible to separate out implementation for exec and composite commands?

Reply:
@johnmcollier can we split this from `getCommand` to `getExecCommandFromDevfile`  and `getCompositeCommandFromDevfile`

Reply:
@adisky Using the command helper functions I wrote, I was able to significantly simplify this function

Reply:
So I was able to avoid needing to create two separate functions 🙂

Reply:
ya, it looks good now :), sorry i hadn't seen revised commit